applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,411 @@
+		return ret;
+	}
+
+	/* Read the header */
+	ret = sd_card_read((char *)&lc3_file_header, &lc3_file_header_size, &f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Read SD card file err: %d", ret);
+		return ret;
+	}
+
+	pcm_frame_size = 2 * lc3_file_header.sample_rate * lc3_file_header.frame_ms / 1000;
+	lc3_playback_cfg.lc3_frames_num =
+		2 * ((lc3_file_header.signal_len_msb << 16) + lc3_file_header.signal_len_lsb) /
+		pcm_frame_size;
+
+	uint8_t pcm_mono_frame[pcm_frame_size];
+
+	sd_card_playback_active = true;
+	for (uint32_t i = 0; i < lc3_playback_cfg.lc3_frames_num; i++) {
+
+		uint8_t lc3_frame[lc3_playback_cfg.lc3_frame_length];
+		size_t lc3_fr_len = lc3_playback_cfg.lc3_frame_length;
+
+		ret = sd_card_read((char *)&lc3_playback_cfg.lc3_frame_length,
+				   &lc3_playback_cfg.lc3_frame_length_size, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			return ret;
+		}
+
+		/* Read the audio data frame to be encoded */
+		ret = sd_card_read((char *)lc3_frame, &lc3_fr_len, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			return ret;
+		}
+		/* Decode audio data frame */
+		ret = sw_codec_lc3_dec_run((char *)lc3_frame, lc3_playback_cfg.lc3_frame_length,
+					   pcm_frame_size, 1, pcm_mono_frame, &pcm_mono_write_size,
+					   false);
+		if (ret < 0) {
+			LOG_ERR("Decoding err: %d", ret);
+			return ret;
+		}
+
+		ret = sd_card_playback_load_ringbuf((char *)pcm_mono_frame, pcm_mono_write_size);
+		if (ret < 0) {
+			LOG_ERR("Load ringbuf err: %d.", ret);
+			return ret;
+		}
+	}
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Close file err: %d", ret);
+		return ret;
+	}
+	sd_card_playback_active = false;
+	return 0;
+}
+
+static int sd_card_playback_play_wav(void)
+{
+	int ret;
+	size_t pcm_wav_mono_frame_size = wav_playback_cfg.frame_duration_ms *
+					 wav_playback_cfg.sample_rate / 1000 *
+					 wav_playback_cfg.bit_depth / 8;
+	pcm_frame_size = pcm_wav_mono_frame_size;
+	uint8_t pcm_wav_mono_frame[pcm_wav_mono_frame_size];
+
+	ret = sd_card_open(playback_file_name, &f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Open SD card file err: %d", ret);
+		return ret;
+	}
+
+	sd_card_playback_active = true;
+	while (sd_card_playback_active) {
+
+		ret = sd_card_read(pcm_wav_mono_frame, &pcm_wav_mono_frame_size, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			return ret;
+		}
+		if (pcm_wav_mono_frame_size < pcm_frame_size) {
+			sd_card_playback_active = false;
+		}
+		/* Wait until there is enough space in the ringbuffer */
+		if (sd_card_playback_active) {
+
+			ret = sd_card_playback_load_ringbuf(pcm_wav_mono_frame,
+							    pcm_wav_mono_frame_size);
+			if (ret < 0) {
+				LOG_ERR("Load ringbuf err: %d.", ret);
+				return ret;
+			}
+		}
+	}
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("SD card close file err: %d", ret);
+		return ret;
+	}
+	sd_card_playback_active = false;
```

Comment:
This shouldn't be have to be set to false here as the `while`-loop won't exit unless this is false.